drivers/interrupt_controller/intc_plic.c

```diff
@@ -291,14 +308,14 @@ static void plic_irq_handler(const struct device *dev)
 		z_irq_spurious(NULL);
 	}
 
-	edge_irq = riscv_plic_is_edge_irq(dev, local_irq);
+	trig_val = riscv_plic_irq_trig_val(dev, local_irq);
 
 	/*
-	 * For edge triggered interrupts, write to the claim_complete register
-	 * to indicate to the PLIC controller that the IRQ has been handled
+	 * For edge-triggered interrupts, write to the claim_complete register
+	 * here to indicate to the PLIC controller that the IRQ has been handled
 	 * for edge triggered interrupts.
 	 */
-	if (edge_irq != 0) {
+	if (trig_val == PLIC_TRIG_EDGE) {
 		sys_write32(local_irq, claim_complete_addr);
```

Comment:
Why do we have different policies to report the IRQ handling completion for EDGE-triggered and LEVEL-triggered modes? Why one is reported earlier while the other one later?

Reply:
Yeah, acking works exactly the same for edge/level irqs, so we should unify this while we're at it. Could you do it @ycsin? Thanks!

The rest LGTM

Reply:
Hey all, I had a look and that seems to be a workaround for Andes' NCEPLIC100, see https://github.com/torvalds/linux/commit/dd46337ca69662b6912bc230d393c4261d126b8f and the discussion [here](https://lore.kernel.org/linux-arm-kernel/CA+V-a8s80shYtUsJZSu7vz_S6ZXtxGkHVi5cgBWO9tr_n+XiJA@mail.gmail.com/t/)

Reply:
Hi @ycsin, 

Interesting! Looks like Andes's PLIC has an issue?

From the thread you've shared:
```
> Is this a feature of this PLIC implementation? Or is that common to
> all PLICs?
>
I think it's a feature of this PLIC implementation and not common to all PLIC's.
```

I think making this WAR behavior default for all the vendors is very risky.

Reply:
Just as a follow-up. Syntacore also has this Triggering Mode register, but since it is not standardized yet, we made it as extension for standard PLIC: https://github.com/syntacore/zephyr/blob/sc/main-3.5/soc/riscv/riscv-privileged/syntacore-scr/common/plic.c

I suppose that other vendors also have some differences in that non-standard functionality implementation and bringing this to a common PLIC driver which is intended to follow the Spec is a "thin ice" path